relay-dynamic-config/src/feature.rs

```diff
@@ -1,30 +1,62 @@
+use std::borrow::Cow;
+
 use serde::{Deserialize, Serialize};
 
 /// Features exposed by project config.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash)]
 pub enum Feature {
     /// Enables ingestion and normalization of profiles.
-    #[serde(rename = "organizations:profiling")]
     Profiling,
     /// Enables ingestion of Session Replays (Replay Recordings and Replay Events).
-    #[serde(rename = "organizations:session-replay")]
     SessionReplay,
     /// Enables data scrubbing of replay recording payloads.
-    #[serde(rename = "organizations:session-replay-recording-scrubbing")]
     SessionReplayRecordingScrubbing,
     /// Enables device.class synthesis
     ///
     /// Enables device.class tag synthesis on mobile events.
-    #[serde(rename = "organizations:device-class-synthesis")]
     DeviceClassSynthesis,
     /// Unused.
     ///
     /// This used to control the initial experimental metrics extraction for sessions and has been
     /// discontinued.
-    #[serde(rename = "organizations:metrics-extraction")]
     Deprecated1,
-
     /// Forward compatibility.
-    #[serde(other)]
-    Unknown,
+    Unknown(String),
+}
+
+impl<'de> Deserialize<'de> for Feature {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let feature_name = Cow::<str>::deserialize(deserializer)?;
+        match feature_name.trim() {
```

Comment:
```suggestion
        match feature_name {
```

If sentry sends something like `"  organizations:profiling   "`, that's a bug we don't have to be robust against.

Reply:
The compiler doesn't like matching `Cow`s and `str`s, so I added `trim` to get the `str`. I changed this to getting the reference instead -- no important downsides, and should be able to spot potential bugs on the project config generation code in sentry.

Reply:
Great!